rpc/core/status.go

```diff
@@ -64,7 +64,12 @@ import (
 //}
 // ```
 func Status() (*ctypes.ResultStatus, error) {
-	latestHeight := blockStore.Height()
+	var latestHeight int64 = -1
+	if consensusReactor.FastSync() {
+		latestHeight = blockStore.Height()
+	} else {
+		latestHeight = consensusState.GetLastHeight()
```

Comment:
this will lock if consensus halts. My solution is here https://github.com/tendermint/tendermint/pull/2009